### PR TITLE
Dg list update to fix diagrams and issues from PE-D

### PR DESCRIPTION
**Pull request by zoranabc201**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -55,14 +55,17 @@ The list feature has the following commands in it -
...
+* `list -pair` This lists all clients and properties that have been paired, in no particular order.
+* `list -pair -short` This lists the -short version of all clients and properties that have been paired, in 
+no particular order
+
+There are 7 different classes, which each inherit from the Command class, and work in similar ways - 
+* They are executed when an object is created in the corresponding Parse class.
+* On execution, they read information about a single client or property and send it to the corresponding 
+display function in Ui
+* The Ui function then displays the necessary information.
+* Finally, it states the number of items present, and the Command object is no longer used.
```
Comment (ngdeqi) [inline docs/DeveloperGuide.md:67]: This description is more condensed and readable. Good job.

**Review by ngdeqi — COMMENTED** [review]
I think the method call in ListSequenceUpdated UML is inaccurate

**Comment by zoranabc201** [comment]
> I think the method call in ListSequenceUpdated UML is inaccurate

What's wrong with it?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -391,42 +394,51 @@ prevent entries retaining within pairingList after it has been deleted from clie
 There are 3 main steps whenever a list command needs to be executed
 * When the user enters any command, it first needs be understood. That is handled by the ParseManager class.
    Next, when the first word entered by the user is determined to be `list`, ParseManager itself then determines
-   the type of list command entered, including the tags. ParseListClient, ParseListProperty and ParseListEverything
-   are checkers to ensure that a valid command has been entered. ParseListClient and ParseListProperty also determine
-   if tags have been entered, and if those tags are valid.
+   the type of list command entered, including the tags. ParseListClient, ParseListProperty, ParseListPair and
+ParseListEverything are checkers to ensure that a valid command has been entered. ParseListClient and ParseListProperty
+also determine if tags have been entered, and if those tags are valid. ParseListPair also checks whether -short 
+has been added or not.
```
Comment (wilsonngja) [inline docs/DeveloperGuide.md:400]: The last sentence seems slightly informal. Perhaps you might want to rephrase?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -391,42 +394,51 @@ prevent entries retaining within pairingList after it has been deleted from clie
...
+        } else if (isListClient) {
+            return new ParseListClient(listCommandTypeAndFlags.get(COMMAND_FLAG_INDEX));
+        } else if (isListEverything && listCommandTypeAndFlags.get(COMMAND_FLAG_INDEX).isEmpty()) {
+            return new ParseListEverything();
+        } else if (isListPairs) {
+            return new ParseListPair(listCommandTypeAndFlags.get(COMMAND_FLAG_INDEX));
+        } else {
+            throw new UndefinedSubCommandTypeException(MESSAGE_INCORRECT_LIST_DETAILS);
+        }
+    }
```
Comment (wilsonngja) [inline docs/DeveloperGuide.md:419]: Would be good if different parts of the code are separated with white space to increase readability.

**Review by wilsonngja — APPROVED** [review]
Overall LGTM. Thank you

**Pull request merged**